frontend/src/components/custom-tools/manage-docs-modal/ManageDocsModal.jsx

```diff
@@ -464,27 +477,72 @@ function ManageDocsModal({
       // If an error occurs while setting custom posthog event, ignore it and continue
     }
 
+    const isFileAlreadyExist = (listOfDocs, fileName) => {
+      const fileNameWithoutExtension = fileName?.replace(/\.[^/.]+$/, ""); // Remove extension from file name
+
+      return [...listOfDocs]?.find(
+        (item) =>
+          item?.document_name?.replace(/\.[^/.]+$/, "") ===
+          fileNameWithoutExtension
+      );
+    };
+
     return new Promise((resolve, reject) => {
+      setResolveUpload(() => resolve); // Save the resolve function
+      setRejectUpload(() => reject); // Save the reject function
+
       const reader = new FileReader();
       reader.readAsDataURL(file);
+
       reader.onload = () => {
         const fileName = file.name;
-        const fileAlreadyExists = [...listOfDocs].find(
-          (item) => item?.document_name === fileName
-        );
-        if (!fileAlreadyExists) {
-          resolve(file);
-        } else {
+        const fileType = file.type;
+        const fileAlreadyExists = isFileAlreadyExist(listOfDocs, fileName);
+
+        // Check if file name already exists
+        if (fileAlreadyExists) {
           setAlertDetails({
             type: "error",
             content: "File name already exists",
           });
           reject(new Error("File name already exists"));
+          return; // Stop further execution
+        }
+
+        // If the file is not a PDF, show the modal for confirmation
+        if (fileType !== "application/pdf") {
+          if (!ConfirmMultiDoc) {
+            setAlertDetails({
+              type: "error",
+              content: "Only PDF files are allowed",
+            });
+          }
```

Comment:
@jagadeeswaran-zipstack if we are allowing multi doc support why we are throwing "Only PDF files are allowed" error?

Reply:
This is for OSS users @vishnuszipstack. Multi Doc support is a cloud feature.